source/docs/software/vision-processing/apriltag/apriltag-pose-estimation.rst

```diff
@@ -0,0 +1,303 @@
+
+   // In your periodic method
+   auto botpose = limelightTable->GetEntry("botpose_wpiblue").GetDoubleArray({});
+   if (!botpose.empty() && botpose[0] != 0.0) {
+      frc::Pose2d visionPose{units::meter_t{botpose[0]}, units::meter_t{botpose[1]},
+                             frc::Rotation2d{units::degree_t{botpose[5]}}};
+      auto latency = limelightTable->GetEntry("tl").GetDouble(0) + limelightTable->GetEntry("cl").GetDouble(0);
+      auto timestamp = frc::Timer::GetFPGATimestamp() - units::millisecond_t{latency};
+
+      poseEstimator.AddVisionMeasurement(visionPose, timestamp);
+   }
+   ```
+
+   ```python
+   from ntcore import NetworkTableInstance
+   from wpilib import Timer
+   from wpimath.geometry import Pose2d, Rotation2d
+
+   limelight_table = NetworkTableInstance.getDefault().getTable("limelight")
+
+   # In your periodic method
+   botpose = limelight_table.getEntry("botpose_wpiblue").getDoubleArray([])
+   if len(botpose) > 0 and botpose[0] != 0.0:
+      vision_pose = Pose2d(botpose[0], botpose[1], Rotation2d.fromDegrees(botpose[5]))
+      latency = limelight_table.getEntry("tl").getDouble(0) + limelight_table.getEntry("cl").getDouble(0)
+      timestamp = Timer.getFPGATimestamp() - (latency / 1000.0)
+
+      pose_estimator.addVisionMeasurement(vision_pose, timestamp)
+   ```
+
+.. note:: Use ``botpose_wpiblue`` or ``botpose_wpired`` based on your alliance color. These provide poses in the correct field coordinate system.
+
+See the `Limelight documentation <https://docs.limelightvision.io/docs/docs-limelight/apis/complete-networktables-api#apriltag-and-3d-data>`_ for the complete NetworkTables API.
+
+## Important Considerations
+
+### Timestamps
+
+It's critical to use the **timestamp from when the image was captured**, not when it was processed. Vision processing introduces latency (typically 20-100ms), and the pose estimator needs the actual capture time to properly fuse the measurement with odometry data.
+
+Most vision libraries provide this timestamp:
+- PhotonVision: ``result.timestampSeconds``
+- Limelight: Calculate from ``tl`` (targeting latency) + ``cl`` (capture latency)
+
+### Standard Deviations
+
+The accuracy of vision measurements varies based on several factors:
+
+- **Distance from tags**: Measurements are less accurate when far from tags
+- **Number of tags**: Seeing multiple tags improves accuracy
+- **Tag ambiguity**: Low-resolution or angled views reduce accuracy
+- **Camera quality**: Higher resolution cameras provide better accuracy
+
+You should scale the standard deviations passed to ``AddVisionMeasurement()`` based on these factors:
+
+.. tab-set-code::
+
+   ```java
+   // Example: Scale standard deviations based on distance and tag count
+   double distance = /* calculate distance to nearest tag */;
+   int tagCount = /* number of tags seen */;
+
+   // More tags = more trust, greater distance = less trust
+   double xyStdDev = 0.5 * Math.pow(distance, 2) / tagCount;
```

Comment:
This formula seems suspect. If you're 1 meter away with 1 tag, it would give a standard deviation of half a meter. But if you're 10 meters away, it would give a standard deviation of 50 meters.